dashboard/modules/job/common.py

```diff
@@ -238,3 +241,77 @@ class JobStopResponse:
 @dataclass
 class JobLogsResponse:
     logs: str
+
+
+class DriverInfo(BaseModel):
+    """A class for recording information about the driver related to the job."""
+
+    id: str = Field(..., description="The id of the driver")
+    ip_address: str = Field(
+        ..., description="The ip address of the node the driver is running on"
+    )
+    pid: str = Field(
+        ..., description="The pid of the worker process the driver is using."
+    )
+
+
+class JobDetails(BaseModel):
+    """
+    Job data with extra details about its driver and its submission.
+    """
+
+    type: str = Field(
+        ..., description='The type of job. Either "submission" or "driver"'
+    )
+    entrypoint: Optional[str] = Field(
+        None, description="The entrypoint command for this job."
+    )
+    id: Optional[str] = Field(
+        None,
+        description="The primary id of a job. The job_id will be the primary id in the "
+        "long term. For backwards compatibility reasons, submission type jobs use "
+        "submission ids as the primary id.",
+    )
+    job_id: Optional[str] = Field(
+        None,
+        description="The job id. An id that is created for every driver that is "
+        "launched in ray. This can be used to fetch data about jobs using ray "
+        "core apis.",
+    )
```

Comment:
It's pretty confusing that this thing is described as "the id for a driver launch in ray" but it's called "**job**_id" :(